src/abstractdataframe/reshape.jl

```diff
@@ -75,17 +76,15 @@ function stack(df::AbstractDataFrame,
     cnames = _names(df)[ints_id_vars]
     push!(cnames, variable_name_s)
     push!(cnames, value_name_s)
-    if variable_eltype <: CategoricalValue{String}
-        nms = names(df, ints_measure_vars)
-        catnms = categorical(nms)
-        levels!(catnms, nms)
-    elseif variable_eltype === Symbol
-        catnms = _names(df)[ints_measure_vars]
+    if variable_eltype === Symbol
+        catnms = PooledArray(_names(df)[ints_measure_vars])
     elseif variable_eltype === String
         catnms = PooledArray(names(df, ints_measure_vars))
     else
-        throw(ArgumentError("`variable_eltype` keyword argument accepts only " *
-                            "`CategoricalValue{String}`, `String` or `Symbol` as a value."))
+        # this covers CategoricalArray{String} in particular,
+        # as copyto! inserts levels in their order of appearance
+        nms = names(df, ints_measure_vars)
+        catnms = copyto!(similar(nms, variable_eltype), nms)
```

Comment:
This relies on `similar` for `CategoricalValue{String}` producing a `CategoricalVector`. Is this something that will be kept being provided in CategoricalArrays.jl? (I guess you intend so, but just making sure.)

Reply:
That's a good question. Given improvements at https://github.com/JuliaLang/julia/pull/37163, we may be able to keep that function in the end. Anyway if CategoricalArrays stops doing that, it will be in a major release so we'll be able to adapt. But yeah if we drop that we'll have to find another approach if we want to drop the CategoricalArrays dependency...

Concretely, for this PR, I could also simply go back to explicitly creating a `CategoricalArray`. That would probably be clearer.

Reply:
Maybe this is what we can do in this PR? I would go for the "smart" path at the moment we decide do drop CategoricalArrays.jl dependency of DataFrames.jl. I will post on #data to ask about it.